### PR TITLE
MLDB-1058: added a "num-threads" parameter to mldb_runner

### PR DESCRIPTION
**Pull request by wsourdeau**

### CODE REVIEW

File: server/mldb_runner.cc [hunk 1]
```diff
@@ -95,6 +95,7 @@ int main(int argc, char ** argv)
     options_description script_options("Script options");
     options_description plugin_options("Plugin options");
 
+    int numThreads(1);
```
Comment (FinchPowers) [inline server/mldb_runner.cc]: You should have left the default to 16 I suppose.

**Comment by jeremybarnes** [comment]
Agreed with Mich.  Having not enough threads causes deadlocks.  It should always have at least 16 threads, and probably 32, unless it auto-scales the number of threads.

**Comment by wsourdeau** [comment]
-1 until MLDB-1058-one-thread-per-loop is mergable.

**Comment by wsourdeau** [comment]
@FinchPowers @jeremybarnes fixed.

File: server/mldb_runner.cc [hunk 2]
```diff
@@ -134,6 +135,7 @@ int main(int argc, char ** argv)
         ("etcd-path", value(&etcdPath),
          "Base path in etcd")
 #endif
+        ("num-threads,t", value(&numThreads), "Number of workers threads")
```
Comment (guyd) [inline server/mldb_runner.cc]: workers -> worker

**Comment by guyd** [comment]
+1

**Comment by FinchPowers** [comment]
If MLDB can deadlock with a single thread I think we should make it impossible to set it to that value.

**Comment by jeremybarnes** [comment]
We're missing an error message (or at least a loud warning) if we set num threads < 4 or so.

**Comment by nicolaskruchten** [comment]
Agreed that we should not allow fewer than 4 threads.

**Comment by wsourdeau** [comment]
@FinchPowers @jeremybarnes @nicolaskruchten I have set the default number of threads to 16, and a strong requirement of 4. The program dies with an error message when using a lesser value.

**Comment by nicolaskruchten** [comment]
+1, please merge

**Pull request merged**